react-components/src/components/Architecture/SettingsButton.tsx

```diff
@@ -101,73 +103,80 @@ export const SettingsButton = ({
 
 function createMenuItem(command: BaseCommand, t: TranslateDelegate): ReactNode {
   if (command instanceof BaseSliderCommand) {
-    return createSlider(command, t);
+    return <SliderComponent key={command.uniqueId} command={command} t={t} />;
   }
   if (command instanceof BaseOptionCommand) {
-    return createDropdownButton(command);
+    return <DropdownButtonComponent key={command.uniqueId} command={command} />;
   }
   if (command instanceof BaseFilterCommand) {
-    return createFilterButton(command);
+    return <FilterButtonComponent key={command.uniqueId} command={command} />;
   }
   if (command.isToggle) {
-    return createToggle(command, t);
+    return <ToggleComponent key={command.uniqueId} command={command} t={t} />;
   }
   if (command instanceof DividerCommand) {
-    return createDivider(command);
+    return <DividerComponent key={command.uniqueId} command={command} />;
   }
   if (command instanceof SectionCommand) {
-    return createSection(command, t);
+    return <SectionComponent key={command.uniqueId} command={command} t={t} />;
   }
-  return createButton(command, t);
+
+  return <ButtonComponent key={command.uniqueId} command={command} t={t} />;
 }
 
-function createDivider(command: BaseCommand): ReactNode {
+function DividerComponent({ command }: { command: BaseCommand }): ReactNode {
   // @update-ui-component-pattern
   const [isVisible, setVisible] = useState(true);
-  const [uniqueId, setUniqueId] = useState(0);
 
   useOnUpdate(command, () => {
     setVisible(command.isVisible);
-    setUniqueId(command.uniqueId);
   });
   // @end
 
   if (!isVisible) {
     return null;
   }
-  return <Menu.Divider key={uniqueId} />;
+  return <Menu.Divider />;
 }
 
-function createSection(command: BaseCommand, t: TranslateDelegate): ReactNode {
+function SectionComponent({
+  t,
+  command
+}: {
+  command: BaseCommand;
+  t: TranslateDelegate;
+}): ReactNode {
   // @update-ui-component-pattern
   const [isVisible, setVisible] = useState(true);
-  const [uniqueId, setUniqueId] = useState(0);
 
   useOnUpdate(command, () => {
     setVisible(command.isVisible);
-    setUniqueId(command.uniqueId);
   });
   // @end
 
   if (!isVisible) {
     return null;
   }
   const label = command.getLabel(t);
-  return <Menu.Section key={uniqueId} label={label} />;
+  return <StyledSectionHeader>{label} </StyledSectionHeader>;
```

Comment:
We switch from `Menu` subcomponents to normal divs, are we sure about these changes? If they were used inside a `Menu`, this will mean a change in behaviour as well.

Reply:
Right, Fredrik caught a similar issue in his risk review on the other, identical PR: https://github.com/cognitedata/reveal/pull/5074#discussion_r2037466261. This is largely a mistake on my end, this change should have been part of the previous PR in this chain, where we switched out the parent component from `Menu` to a custom component. Therefore we no longer want to use `Menu` subcomponents here